packages/@react-spectrum/s2/src/SelectBoxGroup.tsx

```diff
@@ -0,0 +1,451 @@
+              aria-hidden="true">
+              <div
+                className={box({
+                  isSelected: renderProps.isSelected,
+                  isDisabled: renderProps.isDisabled,
+                  size: 'M'
+                } as any)}>
+                {renderProps.isSelected && (
+                  <Checkmark 
+                    size="S" 
+                    className={iconStyles} />
+                )}
+              </div>
+            </div>
+          )}
+          <Provider
+            values={[
+              [IllustrationContext, {
+                size: 'S',
+                styles: illustrationContainer({size: 'S', orientation, isDisabled: renderProps.isDisabled})
+              }],
+              [TextContext, {
+                slots: {
+                  label: {
+                    styles: labelText({orientation, isDisabled: renderProps.isDisabled})
+                  },
+                  description: {
+                    styles: descriptionText({orientation, isDisabled: renderProps.isDisabled})
+                  }
+                }
+              }]
+            ]}>
+            {children}
+          </Provider>
+        </>
+      )}
+    </ListBoxItem>
+  );
+}
+
+/**
+ * SelectBoxGroup allows users to select one or more options from a list.
+ */
+export const SelectBoxGroup = /*#__PURE__*/ forwardRef(function SelectBoxGroup<T>(props: SelectBoxGroupProps<T>, ref: DOMRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, SelectBoxGroupContext);
+
+  let {
+    children,
+    onSelectionChange,
+    selectedKeys: controlledSelectedKeys,
+    defaultSelectedKeys,
+    selectionMode = 'single',
+    orientation = 'vertical',
+    numColumns = 2,
+    gutterWidth = 'default',
+    isDisabled = false,
+    showCheckbox = false,
+    UNSAFE_className,
+    UNSAFE_style
+  } = props;
+
+  const [selectedKeys, setSelectedKeys] = useControlledState(
+    controlledSelectedKeys,
+    defaultSelectedKeys || new Set(),
+    onSelectionChange
+  );
+
+  const childrenArray = React.Children.toArray(children).filter((x) => x);
+  useEffect(() => {
+    if (childrenArray.length > 9) {
+      console.warn('Invalid content. SelectBoxGroup cannot have more than 9 children.');
+    }
+  }, [childrenArray.length]);
+
+  const selectBoxContextValue = useMemo(
+    () => {
+      const contextValue = {
+        allowMultiSelect: selectionMode === 'multiple',
+        orientation,
+        isDisabled,
+        showCheckbox,
+        selectedKeys,
+        onSelectionChange: setSelectedKeys
+      };
+      
+      return contextValue;
+    },
+    [selectionMode, orientation, isDisabled, showCheckbox, selectedKeys, setSelectedKeys]
+  );
+
+  return (
+    <ListBox
+      layout="grid"
+      selectionMode={selectionMode}
+      selectedKeys={selectedKeys}
+      onSelectionChange={setSelectedKeys}
+      className={UNSAFE_className + gridStyles({gutterWidth, orientation}, props.styles)}
```

Comment:
```suggestion
      className={(UNSAFE_className || '') + gridStyles({gutterWidth, orientation}, props.styles)}
```
so we don't get `'undefined'` in our classname string accidentally